src/types.ts

```diff
@@ -137,6 +162,190 @@ type FinalAuthReqsSatisfied<T extends HasFinalAuthorize<any, any>> = IntersectPr
 */
 // Start with the "optionalized" section below
 
+type ObjectValueToAny<T, TKeyProperty extends string | symbol | number> = [
+  T
+] extends [object]
+  ? { [key in keyof T]: any }
+  : Record<TKeyProperty, T>;
+
```

Comment:
In the same spirit as above...
I don't see why we need/want NotObjectCase at all.  Just use T instead of NotObjectCase<...>
- [ ] remove NotObjectCase